airbyte-cdk/bulk/toolkits/load-dlq/src/main/kotlin/io/airbyte/cdk/load/command/dlq/ObjectStorageConfig.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.command.dlq
+
+import io.airbyte.cdk.load.command.aws.AWSAccessKeyConfiguration
+import io.airbyte.cdk.load.command.aws.AWSAccessKeyConfigurationProvider
+import io.airbyte.cdk.load.command.aws.AWSArnRoleConfiguration
+import io.airbyte.cdk.load.command.aws.AWSArnRoleConfigurationProvider
+import io.airbyte.cdk.load.command.object_storage.ObjectStorageCompressionConfiguration
+import io.airbyte.cdk.load.command.object_storage.ObjectStorageCompressionConfigurationProvider
+import io.airbyte.cdk.load.command.object_storage.ObjectStorageFormatConfiguration
+import io.airbyte.cdk.load.command.object_storage.ObjectStorageFormatConfigurationProvider
+import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfiguration
+import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfigurationProvider
+import io.airbyte.cdk.load.command.s3.S3BucketConfiguration
+import io.airbyte.cdk.load.command.s3.S3BucketConfigurationProvider
+import java.io.OutputStream
+
+interface ObjectStorageConfigProvider {
+    val objectStorageConfig: ObjectStorageConfig
+}
+
+sealed interface ObjectStorageConfig {
+    val type: String
+}
+
+class DisabledObjectStorageConfig : ObjectStorageConfig {
+    override val type: String = "None"
+}
```

Comment:
Why not just allow it to be nullable?

Reply:
This is to enable `@Requires(bean = ObjectStorageConfig::class, beanProperty = "type", notEquals = "None")` further down on the `@Named("dlqPipelineSteps")`.
Micronaut doesn't deal well with the nullables for the `@Singleton` return types.